### PR TITLE
Core: Replace SSE with websockets

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: #16489

## What I did

 - Replace the use of SSE and a custom client with the `ws` library and the existing `channel-websocket`.
 - Rudimentary setup of a global system for server->{manager,preview} communication.

## How to test

- [x] Is this testable with Jest or Chromatic screenshots?

Try adding/removing stories in `react-ts`

### CODE REVIEW

File: lib/builder-webpack4/src/preview/virtualModuleModernEntry.js.handlebars [hunk 1]
```diff
@@ -1,39 +1,47 @@
...
 
+const { SERVER_CHANNEL_URL } = global;
+
 const getProjectAnnotations = () =>
-  composeConfigs([
-    {{#each configs}}
-    require('{{this}}'),
-    {{/each}}
-  ]);
+composeConfigs([
```
Comment (shilman) [inline lib/builder-webpack4/src/preview/virtualModuleModernEntry.js.handlebars:15]: @tmeasday is this indentation intentional?

File: lib/builder-webpack4/src/preview/virtualModuleModernEntry.js.handlebars [hunk 2]
```diff
@@ -1,39 +1,47 @@
...
-    // importFn has changed so we need to patch the new one in
-    preview.onStoriesChanged({ importFn });
-  });
-
-  module.hot.accept([{{#each configs}}'{{this}}',{{/each}}], () => {
-    // getProjectAnnotations has changed so we need to patch the new one in
-    preview.onGetProjectAnnotationsChanged({ getProjectAnnotations });
-  });
-}
+module.hot.accept('./{{storiesFilename}}', () => {
```
Comment (shilman) [inline lib/builder-webpack4/src/preview/virtualModuleModernEntry.js.handlebars:38]: also indentation?

File: lib/builder-webpack5/src/preview/virtualModuleModernEntry.js.handlebars [hunk 3]
```diff
@@ -1,39 +1,47 @@
...
-    // importFn has changed so we need to patch the new one in
-    preview.onStoriesChanged({ importFn });
-  });
-
-  module.hot.accept([{{#each configs}}'{{this}}',{{/each}}], () => {
-    // getProjectAnnotations has changed so we need to patch the new one in
-    preview.onGetProjectAnnotationsChanged({ getProjectAnnotations });
-  });
-}
+module.hot.accept('./{{storiesFilename}}', () => {
```
Comment (shilman) [inline lib/builder-webpack5/src/preview/virtualModuleModernEntry.js.handlebars:38]: same

File: lib/core-server/src/utils/server-address.ts [hunk 4]
```diff
@@ -15,3 +15,7 @@ export const getServerPort = (port: number) =>
     logger.error(error);
     process.exit(-1);
   });
+
+export const getServerChannelUrl = (port: number, { https }: { https?: boolean }) => {
+  return `${https ? 'wss' : 'ws'}://localhost:${port}/`;
```
Comment (shilman) [inline lib/core-server/src/utils/server-address.ts:20]: is it always localhost?

**Review by shilman — APPROVED** [review]
Much cleaner! Just a few formatting nits

File: lib/builder-webpack4/src/preview/virtualModuleModernEntry.js.handlebars (hunk 1 above)
Comment (tmeasday) [reply lib/builder-webpack4/src/preview/virtualModuleModernEntry.js.handlebars:15]: Ughh, no, sorry not sure what happened.

File: lib/core-server/src/utils/server-address.ts (hunk 4 above)
Comment (tmeasday) [reply lib/core-server/src/utils/server-address.ts:20]: I followed the pattern on `getServerAddresses` above. I'm not totally sure it is always `localhost` but let's start with that and see.

**Pull request merged**